CaseChange/CaseChange.glyphsPalette/Contents/Resources/plugin.py

```diff
@@ -86,22 +86,37 @@ def changeCaseCallback_(self, sender):
 		if tab and tab.text:
 			f = self.windowController().documentFont()
 			previousGlyphTypeIsSeparator = True
-			currentLayers = tab.layers
-			# Text to be used to replace the current text
-			newText = ""
+		
+			# change the whole tab
+			if tab.textRange == 0:
```

Comment:
I didn't know about this `textRange` property. I'm currently focused on other programming projects not related to type design at this moment so I'll need to catch up on Glyphs 3 and its API.

Maybe @mekkablue can give some feedback because I would like to keep this plugin working for Glyphs 2 for a bit longer.

Reply:
It seem to work fine in Glyphs 2 as well!